metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -67,10 +126,93 @@ class PostgresSource(SQLAlchemySource):
     - Table, row, and column statistics via optional SQL profiling
     """
 
-    def __init__(self, config, ctx):
+    def __init__(self, config: PostgresConfig, ctx: PipelineContext):
         super().__init__(config, ctx, "postgres")
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = PostgresConfig.parse_obj(config_dict)
         return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]:
+        yield from super().get_workunits()
+
+        if self.config.include_view_lineage:  # type: ignore
+            yield from self._get_view_lineage_workunits()
+
+    def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
+        url = self.config.get_sql_alchemy_url()
+        engine = create_engine(url, **self.config.options)
+
+        data: List[ViewLineageEntry] = []
+
+        with engine.connect() as conn:
+            results: CursorResult = conn.execute(VIEW_LINEAGE_QUERY)
+            if results.returns_rows is False:
+                return None
+
+            for row in results:
+                data.append(ViewLineageEntry.parse_obj(row))
+
+        if len(data) == 0:
+            return None
+
+        lineage_elements: Dict[str, List[str]] = {}
```

Comment:
You can make the "key" a tuple of schema, view_name so that you don't need the `":::::".join` later on
```suggestion
        lineage_elements: Dict[Tuple[str, str], List[str]] = {}
```

Reply:
of course, yes, sorry i have been mostly coding in Go for the last several years and forgot that tuples were hashable valid keys.